acceptancetests/clean_resources.py

```diff
@@ -1,60 +1,143 @@
+        logging.info(
+            'Unable to clean {} groups from {}'.format(
+            str(len(unclean)), region))
+    except:
+        logging.info(
+            'Unable to clean non-instance groups in region {} from\n{}'.format(
+            region,
+            '\n'.join(non_instgrp.keys())))
+    return unclean
+
+
+def remove_security_groups(unclean, grp, non_instgrp, substrate, region):
+    """Remove non-instance security groups by group name"""
+    for group_id in unclean:
+        logging.info(
+            'Cannot delete {} from {}'.format(
+            grp[group_id], region))
+        non_instgrp.pop(group_id, None)
+    try:
+        substrate.destroy_security_groups(non_instgrp.values())
+        logging.info(
+            '{} non-instance groups have been deleted from {}\n{}'.format(
+            len(non_instgrp.values()),
+            region,
+            '\n'.join(non_instgrp.values())))
+    except:
+        logging.info(
+            'Failed to delete groups {} from {}'.format(
+            non_instgrp.values(), region))
 
 
 def clean(args):
     env = SimpleEnvironment.from_config(args.env)
-    selected_regions = get_regions(args, env)
+    selected_regions = get_regions()
+    logging.info(
+        'The target regions for cleaning job are \n{}'.format(
+        '\n'.join(selected_regions)))
     for region in selected_regions:
         with AWSAccount.from_boot_config(env, region=region) as substrate:
-            logging.info('Cleaning resources in {}.'.format(substrate.region))
-            all_groups = dict(substrate.iter_security_groups())
-            instance_groups = dict(substrate.iter_instance_security_groups())
-            non_instance_groups = dict((k, v) for k, v in all_groups.items()
-                                       if k not in instance_groups)
-            try:
-                unclean = substrate.delete_detached_interfaces(
-                    non_instance_groups.keys())
-                logging.info('Unable to delete {} groups'.format(len(unclean)))
-            except:
-                logging.info('Unable to delete non-instance groups {}'.format(non_instance_groups.keys()))
-            for group_id in unclean:
-                logging.debug('Cannot delete {}'.format(all_groups[group_id]))
-            for group_id in unclean:
-                non_instance_groups.pop(group_id, None)
-            try:
-                substrate.destroy_security_groups(non_instance_groups.values())
-            except:
-                logging.debug('Failed to delete groups {}'.format(non_instance_groups.values()))
+            if substrate is None:
+                logging.info(
+                    'Skipping {}, substrate object returns NoneType'.format(region))
+                continue
+            logging.info(
+                'Cleaning resources in {}.'.format(substrate.region))
+            all_groups, non_instance_groups = get_security_groups(
+                grp=substrate.iter_security_groups(),
```

Comment:
This is a little confusing, passing in a generator and having the function create a dict from it, having the function take a dict is much more straight forward and ensures the dependency of knowledge flows downwards (i.e. not having to know the internals of this function creates a dict from it, just knowing to pass a dict is enough and better as the information of how the function works stays at the boundary (i.e. the function signature)).